src/main/java/seedu/mealcompanion/command/misc/RecipePossibleCommand.java

```diff
@@ -49,21 +55,26 @@ private boolean canMakeRecipe(Recipe recipe, IngredientList fridgeIngredients) {
     }
 
     /**
-     * List all recipes that can be made using ingredients that are available.
+     * List all recipes that can be made using ingredients that are available and do not contain any allergens.
      *
-     * @param mealCompanionSession the MealCompanionSession containing the list of recipes and ingredients
+     * @param mealCompanionSession the MealCompanionSession containing the list of recipes, ingredients, and allergens
      */
     @Override
     public void execute(MealCompanionSession mealCompanionSession) {
+        mealCompanionSession.getUi().printMessage("Please enter allergens (comma-separated): ");
+        String allergensStr = mealCompanionSession.getUi().getNextCommandString();
+        List<String> allergens = Arrays.asList(allergensStr.split(","));
+        mealCompanionSession.setAllergens(allergens);
```

Comment:
Yes I believe it would be better if a separate command was created to store the allergens. This way the user will not need to repeatedly key in their allergens every time they attempt to run this command. Can also consider to store the allergens locally like our list of ingredients so that user don't have to re input their allergens every time they start the program again.